core/trino-main/src/main/java/io/trino/memory/ClusterMemoryManager.java

```diff
@@ -102,7 +103,7 @@
     private final JsonCodec<MemoryInfo> memoryInfoCodec;
     private final DataSize maxQueryMemory;
     private final DataSize maxQueryTotalMemory;
-    private final LowMemoryKiller lowMemoryKiller;
+    private final List<LowMemoryKiller> lowMemoryKillers;
```

Comment:
It's a bit weird to have multiple low memory killers, unless we plan to support pipelined execution and fault tolerant execution in the same cluster.

Reply:
I agree that if properly used cluster only one should trigger. But currently, we are not forcibly forbidding users from running both types of workloads in the same cluster, so we need to handle that in code I think.

Reply:
Does it make sense to remove the session property for retry-policy (or at least don't allow hybrid mode) right now? cc @arhimondr

Reply:
Actually I don't mind removing the session property since currently we don't support mixed workloads. Then based on the `retry-policy` value we may inject the killer we need to keep it simple. However we would still probably need two configuration properties, one that defines what killer to use for killing tasks and the other one for queries as they are going to have different defaults. We can pick one based on the `retry-policy` then. It feels like it would make things simpler.

Reply:
It could be done - but we would need to rework testing. In FTE tests we currently run query with and  without retries. We switch the mode using session property. We would need to migrate approach to have two query runners - can be done but tests will be a bit longer and would require more resources (we need two query runners up and running). 
I will see how easy it is to change it in a separate PR.